sdk/dotnet/Pulumi.Automation/LocalWorkspace.cs

```diff
@@ -34,13 +35,28 @@ public sealed class LocalWorkspace : Workspace
         private readonly LocalSerializer _serializer = new LocalSerializer();
         private readonly bool _ownsWorkingDir;
         private readonly Task _readyTask;
+        private static readonly SemVersion _minimumVersion = SemVersion.Parse("2.21.0");
```

Comment:
Is this something that you guys are intending will be updated regularly?

Reply:
Only when there is a change to the automation SDK that is dependent on a change in the CLI (e.g. the `--page-size` flag being added to history). When we release 3.0 this will be bumped to `3.0.0` (more context: https://github.com/pulumi/pulumi/pull/6577#discussion_r599238753). Recognizing that this can be disruptive we will most likely avoid this outside of major version bumps.